src/lib/libfs.js

```diff
@@ -1780,7 +1780,7 @@ FS.staticInit();`;
             }
 
             xhr.send(null);
-            if (!(xhr.status >= 200 && xhr.status < 300 || xhr.status === 304)) throw new Error("Couldn't load " + url + ". Status: " + xhr.status);
+            if (!(xhr.status >= 200 && xhr.status < 300 || xhr.status === 304)) abort("Couldn't load " + url + ". Status: " + xhr.status);
```

Comment:
You mentioned in the PR description you used `assert` too, but looks like quite a few of conditional aborts like this one could still be replaced with assert.

Just for clarity of review, did these get missed or is it something you want to leave out of scope of this PR?

Reply:
The problem with assert is it that its only available in debug builds (or at least that is the idea).

Reply:
If there are `abort`s list this that are already inside `#if ASSERTIONS` then yes the should be converted to `assert`.    Did you spot any?

Reply:
Ah ok.

It's something I was wondering about from time to time - currently each assert is always wrapped in manual `#if ASSERTIONS`...

If it's meant to be only available in debug builds, maybe it's worth making it conditional in acorn-optimizer to avoid repetitions, or, vice versa, if we want to explicitly wrap only certain assertions in `#if`, then maybe the function definition should always be available so we could use it in all these other places too.

Reply:
The `assert` function is, in theory, only defined when `ASSERTIONS` setting set.  However, in practice this is only true in `MINIMAL_RUNTIME` or `STRICT` modes today, because historically we have not been great a wrapping the calls in `#if ASSERTIONS` in all places.

Regarding using acorn passed instead of our preprocessor shit, that is something I've looked into in the past but haven't managed to land yet.  My latest attempt was here: https://github.com/emscripten-core/emscripten/pull/22166.